teps/0129-multiple-tekton-instances-per-cluster.md

```diff
@@ -95,40 +95,181 @@ bundled in the Tekton webhook as well the remote resolution controller.
+
+### Compatibility Matrix
+
+Tekton API compatibility policy defines that no backwards-incompatible changes are
+allowed on the `v1` version of CRDs. As a consequence of that, the most recent set of
+shared components (CRD, conversion webhook, RBAC) should be able to handle the `v1`
+CRD from all versions installed in a cluster.
+
+Versions of Tekton that serve an `alpha` level API on any resources can be used in
+a multi-installed cluster, however the resource at `alpha` level may be broken.
+
+Versions of Tekton that serve a `beta` level API on any resources can be used in
+a multi-installed cluster, however the resource at `beta` level may or may not work,
+depending on whether the the most recent of the `beta` APIs is compatible with the
+other ones installed. The compatibility matrix will keep track of
+backwards-incompatible changes in `beta` level APIs, to help users estimate which
+versions are compatible. 
+
+The installation of multiple `v1beta1` versions of Tekton in the same cluster will
+**not** be tested in the upstream CI.
+
+### Custom Conversion Webhook
+
+In some cases, users may develop custom conversion controllers for resources and
+versions that are not supported by the Tekton community. Such controllers can
+follow the same structure as the community one, they must include the associated
+CRDs and RBAC and must be configured so to avoid conflict with the community
+conversion controller.
+
+For instance a conversion controller between `v1alpha1.Run` and `v1beta1.CustomRun`
+could be deployed, along with the `v1alpha1.Run` CRD, to allow a version of
+Tekton Pipelines that works with `v1alpha1.Run` to co-exist with a more recent version
+that works with `v1beta1.CustomRun`.
+
+### Operator
+
+The operator will not orchestrate installation, updates or uninstallation of Tekton
+in a cluster with multiple versions installed. It will expose configuration options
+to allow users to orchestrate them.
+
+The operator will be enhanced with the following options:
+- configuration option to skip the installation of shared component
+- configuration option to update the installation of shared components if a previous
+  version is found in the cluster
+- configuration option to not-uninstall shared components upon uninstallation
+
+### Tekton Lifecycle
+
+Multiple installations of Tekton within the same cluster share some components.
+As a consequence, in case of multiple installations, it is not entirely possible to add,
+update and remove a specific installations of Tekton with no impact and no knowledge of
+the other installations.
+
+This TEP will provide documentation with guidelines on how to handle the lifecycle of
+different Tekton installations but it will **not** implement automation to manage them.
+The documentation will example identifies sequences like the one below, and describe what
+should happen for each of the shared components:
+
+Sequence 1:
```

Comment:
Does this sequence infer support for any number of Tekton controllers in a cluster? Would this also cover support for the same version but serving different APIs?

Reply:
> Would this also cover support for the same version but serving different APIs?

I am not sure I see what this would mean ?